src/components/name-input.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  NameInputView,
+  TextInputWrapper,
+  NextButton,
+  ButtonText,
+  NameInputText,
+  WhiteArrowIcon,
+} from '../styles/pages/welcome';
+import ArrowIcon from '../icons/white-arrow.png';
+
+export const NameInput = ({ nextPage }) => (
+  <NameInputView style={{ flex: 1, flexDirection: 'row' }}>
+    <TextInputWrapper>
+      <NameInputText placeholder="What&apos;s your name?" />
+    </TextInputWrapper>
+    <NextButton onPress={() => nextPage()}>
```

Comment:
What happens if we change this to `onPress={nextPage}`? Unless the handler cares about the arity/args, we should be able to get away with not creating an inline callback. Not necessary if it's not a quick change.